FormalConjectures/ErdosProblems/44.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE/2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 44: Extending Sidon Sets
+
+*Reference:* [erdosproblems.com/44](https://www.erdosproblems.com/44)
+-/
+
+open Function Set
+
+/-- The maximum size of a Sidon set in `{1, ..., N}`. -/
+noncomputable def maxSidonSetSize (N : ℕ) : ℕ :=
+  sSup {(A.card) | (A : Finset ℕ) (_ : A ⊆ Finset.Icc 1 N) (_ : IsSidon A.toSet)}
+
+/-- The maximum size of a Sidon set in `{1, ..., N}` is less than or equal to `√N + 1`. -/
+@[category API, AMS 5 11]
+theorem maxSidonSetSize_bound (N : ℕ) (hN : 1 ≤ N) :
+    maxSidonSetSize N ≤ N.sqrt + 1 := by
+  sorry
```

Comment:
Do you know how hard / easy this statement is? I would not use the tag `category API` but maybe `category undergraduate`, `category graduate` or `category research solved` depending on its difficulty.